pkg/labels/labels.go

```diff
@@ -632,14 +641,24 @@ func ParseLabel(str string) *Label {
 			lbl.Value = next[i+1:]
 		}
 	}
-	return &lbl
+	return lbl
 }
 
 // ParseSelectLabel returns a selecting label representation of the given
 // string. Unlike ParseLabel, if source is unspecified, the source defaults to
 // LabelSourceAny
 func ParseSelectLabel(str string) *Label {
-	lbl := ParseLabel(str)
+	lbl := parseSelectLabel(str, ':')
+
+	return &lbl
+}
+
+// parseSelectLabel returns a selecting label representation of the given
+// string by value.
+// For Cilium format 'delim' must be passed in as ':'
+// For k8s format 'delim' must be passed in as '.'
+func parseSelectLabel(str string, delim byte) Label {
```

Comment:
Same here, why not returning a Label pointer?

Reply:
See above